_source/_posts/2024-08-15-otp-over-sms.md

```diff
@@ -0,0 +1,178 @@
+
+Moving to a service provider minimizes the technical requirements, though there's still vendor management and monitoring.
+
+## Designing a DIY Hook
+
+The first step in implementing a telephony hook is finding a vendor. There are at least three essential criteria:
+
+* Send messages to the desired regions
+
+* Meet reliability requirements, especially when handling failover
+
+* Allow the desired volume of messages
+
+That last point is because some vendors limit the volume for smaller or unknown companies.
+
+The server you write for the telephony hook uses the information received from Okta to construct a message request to your vendor. The status of the message also needs to be communicated back to Okta. Sometimes, this requires translating the data from the telephony provider into the JSON format expected by Okta.
+
+### Handling failover to Okta
+
+Another case you must handle is a failover to Okta. Failover happens when something goes wrong with your telephony hook. Okta takes over sending the message, but the number of messages is heavily rate-limited. The only way to determine if the message was sent is by searching the logs to see when sends started failing. Your messages may never arrive.
+
+There are two triggers for failover: your telephony hook returns a "failed" status to Okta, or a three-second timeout passes. 
+
+You can prevent failover by always returning a successful result or requesting Okta to disable failover for your organization. However, doing so means that you must handle failed message sends. That requires more complex server code and possibly multiple vendors.
+
+## Vendors
+
+The kind of vendors you need depends on your approach. Below are a few possibilities. Some are [recommendations from Okta](https://www.okta.com/okta-telecommunication-services/), and others are suggestions. No matter what, make sure that the vendor meets your criteria.
+
+### Telephony providers
+
+Here are some vendors you can use to implement the hook in-house or with a consultant.
+
+* [Telesign](https://www.telesign.com/okta-telesign)
+
+* [Twillio](https://pages.twilio.com/Twilio-Messaging-Okta.html)
+
+* [Vonage](https://www.vonage.com/communications-apis/campaigns/okta/)
+
+### Consultants
+
+Many consulting companies can implement the hook for you. Another option is to use [Okta professional services](https://www.okta.com/services/professional-services/).
+
+### Services
+
+Some services deliver the SMS for you. That can include handling unavailable telephony vendors, resends, and other issues. Adding a service usually requires only adding a URL for the telephony hook.
+
+Services include:
+
+* [AWS Pinpoint](https://aws.amazon.com/pinpoint/)
+
+* [BeyondID](https://beyondid.com/solutions/sms-for-okta/)
+
+* [Twilio Verify](https://www.twilio.com/docs/verify/api)
+
+## What Next? 
```

Comment:
```suggestion
## What's next? 
```

I still recommend working in keywords though... But headers are sentence case per the writing guidelines.